### PR TITLE
Manage cohorts

### PR DESCRIPTION
**Pull request by kretep**

### CODE REVIEW

File: src/app/components/cohort-view/cohort-view.service.spec.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+/* jshint undef: false */
+
+fdescribe('CohortViewService', function () {
```
Comment (rnugraha) [inline src/app/components/cohort-view/cohort-view.service.spec.js]: change this to normal describe

File: src/app/components/cohort-view/cohort-view.directive.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * @memberof transmartBaseUi
+ * @ngdoc directive
+ * @name cohortView
+ */
+angular.module('transmartBaseUi')
+    .directive('cohortView', function () {
+        return {
+            restrict: 'E',
+            templateUrl: 'app/components/cohort-view/cohort-view.tpl.html',
+            controller: ['$scope', '$timeout', '$q', 'CohortViewService', 'AlertService',
```
Comment (rnugraha) [inline src/app/components/cohort-view/cohort-view.directive.js]: Could you also provide ng-docs for controller and its functions?

File: src/app/components/cohort-view/cohort-view.service.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    /** Retrieves the list of cohorts.
+     *
+     * @returns {*} A promise that resolves when the list has been retrieved or is
+     *              rejected in case of an error.
+     * @memberof CohortViewService
+     */
+    service.getCohorts = function() {
+        var deferred = $q.defer();
+
+        // TODO: allow cohorts to be retrieved from other sources than the master endpoint?
```
Comment (rnugraha) [inline src/app/components/cohort-view/cohort-view.service.js]: can you help create JIRA ticket?

File: src/app/components/cohort-view/cohort-view.service.spec.js (hunk 1 above)
Comment (rnugraha) [inline src/app/components/cohort-view/cohort-view.service.spec.js]: please change fdescribe

File: src/app/components/cohort-view/cohort-view.directive.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                    var ctrl = this;
+
+                    ctrl.gridOptions = CohortViewService.options;
+
+                    /** Loads the list of cohorts.
+                     */
+                    ctrl.loadCohorts = function() {
+                        CohortViewService.getCohorts().then(function (cohorts) {
+                            $scope.cohorts = cohorts;
+                        });
```
Comment (rnugraha) [inline src/app/components/cohort-view/cohort-view.directive.js]: In    CohortViewService.getCohorts you defined deferred.reject. What is your plan to it I didnt see you use it here?

File: src/app/components/cohort-view/cohort-view.directive.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                        }, function(error) {
+                            AlertService.add('danger', 'Failed to delete cohort "' + cohort.name + '"');
+                            deferred.reject(error);
+                        });
+                        return deferred.promise;
+                    };
+
+                    /** Deletes the cohorts currently selected in the cohort grid.
+                     */
+                    ctrl.deleteSelectedCohorts = function() {
```
Comment (rnugraha) [inline src/app/components/cohort-view/cohort-view.directive.js]: minor comments .. what's difference between can we stick to one verb .. remove or delete?

File: src/app/components/cohort-view/cohort-view.service.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    };
+
+    /** Removes the specified cohort.
+     * @param cohort The cohort to be deleted.
+     * @returns {*} A promise that is resolved on successful completion of deletion,
+     *              or rejected in case of an error.
+     * @memberof CohortViewService
+     */
+    service.removeCohort = function(cohort) {
+        return cohort.remove();
```
Comment (rnugraha) [inline src/app/components/cohort-view/cohort-view.service.js]: what if cohort is undefined or other object that does not have remove()

**Comment by berndvdveen** [comment]
@kretep Please adapt the cohort-view.mock file to be smaller and representative of the situational data used in testing.

File: src/app/components/cohort-view/cohort-view.service.js (hunk 3 above)
Comment (kretep) [reply src/app/components/cohort-view/cohort-view.service.js:37]: Added ticket TRANSUI-210

**Comment by kretep** [comment]
@rnugraha @berndvdveen @beaugogh I've processed the comments. Please have another look.

**Pull request merged**